source/standards/performance-testing.html.md.erb

```diff
@@ -0,0 +1,99 @@
+
+### Inform people
+
+Inform people affected by your test. For example, tell hosting providers or developers planning to use the test environment of your test’s date, time and duration. Make sure everyone involved in the test agrees to its conditions.
+
+### Build your test
+
+You do not need to fully hand code your tests because most load testing tools like [Gatling](https://gatling.io/), include scenario recorders. Some test scenarios may still need manual preparation, for example, providing lists of test users or specific search terms.
+
+Gatling is a popular load testing tool used by many teams at GDS including [GOV.UK Pay](https://www.payments.service.gov.uk/), [GOV.UK Notify](https://www.notifications.service.gov.uk/) and [Digital Marketplace](https://www.digitalmarketplace.service.gov.uk/).
+
+## Configure your test environment
+
+Test environments must resemble your production environment as far as possible so tests are accurate. Isolate test environments to stop them affecting your production environment.
+
+Consider how to [monitor service performance](https://gds-way.cloudapps.digital/standards/monitoring.html) during the load test. Your service will likely already have dashboards and graphs set up for common metrics such as CPU and memory utilization. Decide if your current metrics are sufficient or if you need to turn on additional logging or configure profiling traces.
+
+Perform practice runs of your tests to make sure they work as expected. For example, you may encounter tooling issues related to [Cross-Site Request Forgery (CSRF)](https://www.owasp.org/index.php/Cross-Site_Request_Forgery_(CSRF)) in your test environment.
+
+## Run your tests
+
+Start with a low level of concurrent users and increase them to your anticipated maximum. If there are no problems, increase the amount of traffic until you reach a breaking point in your system.
+
+## Document and share your results
+
+When testing your service, record:
+
+- the load when the service broke
+- how the service broke and what went wrong
+- the changes introduced to improve performance
+
+If your service meets your capacity requirements, document its upper limits and communicate them to the service owner.
+
+## Analyse your results
+
+Testing tools should record the number of concurrent users, RPS, HTTP response codes and timings. These reports help determine your service’s maximum supported capacity. If your service cannot meet your requirements or you find problems, you’ll need to diagnose them.
+
+Common problems like inefficient database queries can be fixed by introducing lazy loading and adding or altering indexes. Or when [Webserver workers](https://developers.google.com/web/fundamentals/primers/service-workers/) are blocked waiting for API responses, adjusting thread count settings can help depending on your specific use cases and technology.
```

Comment:
This bit isn't quite right. Service workers are a client side thing and I don't think there will be a wiki article on how to configure webserver workers/threads as differs per tool. How about altering this paragraph to something like:

Common problems like inefficient database queries can be fixed by introducing lazy loading and adding or altering indexes. If your webserver stops or is slow to respond, it may be running out of available workers. This is a common issue when you're sending lots of requests to APIs and the workers become blocked awaiting a response. Adjusting webserver configuration can help depending on your specific application and technology.